openssl-dynamic/src/main/c/error.c

```diff
@@ -49,6 +49,19 @@ void tcn_ThrowException(JNIEnv *env, const char *msg)
 
 }
 
+void tcn_ThrowIllegalArgumentException(JNIEnv *env, const char *msg)
+{
+    jclass javaExceptionClass;
+
+    javaExceptionClass = (*env)->FindClass(env, "java/lang/IllegalArgumentException");
```

Comment:
Also note that I could have cached this class but I didnt feel we really need to as this should not happen frequently